docs/en/styling-and-theming/supplemental.md

```diff
@@ -0,0 +1,642 @@
+
+Dojo differentiates between two types of styling, each representing a different granularity of styling concerns within an enterprise web application:
+
+-   [Structural styles](#structural-widget-styling) (_granularity:_ per-widget)
+    -   The minimum styles necessary for a widget to function. These are localized to the particular widget they are needed for.
+-   [Presentational styles](#theming-a-dojo-application) (_granularity:_ application-wide)
+    -   Styles - usually visual - that may be changed without impacting widget functionality. These can cut across several widgets and usually provide a consistent theme for all widgets used within an application.
+
+## Structural widget styling
+
+Dojo leverages [CSS Modules](https://github.com/css-modules/css-modules) to provide all of the flexibility of CSS, but with the additional benefit of localized classes to help prevent inadvertent styling collisions across a large application. Dojo also makes use of [typed CSS modules](https://github.com/Quramy/typed-css-modules), allowing widgets to import their CSS modules similar to any other TypeScript module, and refer to specific CSS class names at design-time via IDE autocompletion.
+
+The CSS module file for a widget should have a `.m.css` extension, and by convention is usually named the same as the widget it is associated with.
+
+### Example
+
+Given the following CSS module file for a widget:
+
+> src/styles/MyWidget.m.css
+
+```css
+.myWidgetClass {
+	font-variant: small-caps;
+}
+
+.myWidgetExtraClass {
+	font-style: italic;
+}
+```
+
+This stylesheet can be used within a corresponding widget as follows:
+
+> src/widgets/MyWidget.ts
+
+```ts
+import WidgetBase from '@dojo/framework/widget-core/WidgetBase';
+import { v } from '@dojo/framework/widget-core/d';
+
+import * as css from '../styles/MyWidget.m.css';
+
+export default class MyWidget extends WidgetBase {
+	protected render() {
+		return v('div', { classes: [css.myWidgetClass, css.myWidgetExtraClass] }, ['Hello from a Dojo widget!']);
+	}
+}
+```
+
+Similarly, if using TSX widget syntax:
+
+> src/widgets/MyTsxWidget.tsx
+
+```tsx
+import WidgetBase from '@dojo/framework/widget-core/WidgetBase';
+import { tsx } from '@dojo/framework/widget-core/tsx';
+
+import * as css from '../styles/MyWidget.m.css';
+
+export default class MyTsxWidget extends WidgetBase {
+	protected render() {
+		return <div classes={[css.myWidgetClass, css.myWidgetExtraClass]}>Hello from a Dojo TSX widget!</div>;
+	}
+}
+```
+
+When inspecting the CSS classes of these sample widgets in a built application, they will not contain `myWidgetClass` and `myWidgetExtraClass`, but rather obfuscated CSS class names similar to `MyWidget-m__myWidgetClass__33zN8` and `MyWidget-m__myWidgetExtraClass___g3St`.
```

Comment:
I believe we might just use an 8 char hash now?

Reply:
Nope we don't, we use a more descriptive hash for both dev and dist